src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -266,11 +267,5 @@ public override int GetHashCode()
             readonly string propertyName;
             readonly object propertyValue;
         }
-
-        static class DefaultSagaDataTask<TSagaData>
```

Comment:
Is this something that should be removed? This was here for creating and caching a single Task, and without this it means we are allocating a new task for each call.

Reply:
We just return null now? https://github.com/Particular/NServiceBus/pull/4923/files#diff-17e8d72500928dcb599bc890835658eeR59 (so no need for the cache?)

Reply:
`Task.FromResult` creates a new `Task`, which we're setting the result value to `null`, so we're still allocating a `Task` per call now compared to before.

https://referencesource.microsoft.com/#mscorlib/system/threading/Tasks/Task.cs,5517

Reply:
Got it, in that case I guess need to add it back

Reply:
I've added it back